account_statement_import_online_ponto/models/online_bank_statement_provider_ponto.py

```diff
@@ -156,34 +156,32 @@ def _ponto_get_transaction(self, account_id, date_since, date_until):
             response = requests.get(
                 page_url, params=params, headers=self._ponto_header()
             )
-            if response.status_code == 200:
-                if params.get("before"):
-                    params.pop("before")
-                data = json.loads(response.text)
-                links = data.get("links", {})
-                if page_next:
-                    page_url = links.get("next", False)
-                else:
-                    page_url = links.get("prev", False)
-                transactions = data.get("data", [])
-                if transactions:
-                    current_transactions = []
-                    for transaction in transactions:
-                        date = self._ponto_date_from_string(
-                            transaction.get("attributes", {}).get("executionDate")
-                        )
-                        if date_since <= date < date_until:
-                            current_transactions.append(transaction)
-
-                    if current_transactions:
-                        if not page_next or (page_next and not latest_identifier):
-                            latest_identifier = current_transactions[0].get("id")
-                        transaction_lines.extend(current_transactions)
-            else:
+            if response.status_code != 200:
                 raise UserError(
                     _("Error during get transaction.\n\n%s \n\n %s")
                     % (response.status_code, response.text)
                 )
+            if params.get("before"):
```

Comment:
Why changing this order? Please try to minimize diff

Reply:
I thought it was cleaner, saving a lot of indented code and getting rid of an `else` branch. But if you still object I will change it back.